### PR TITLE
Airborne Aero Artillery Fire (Bug #205)

### PR DESCRIPTION
**Pull request by mkerensky**
This fixes bug #205 , correcting a number of issues within the Artillery Bay Indirect Handler that threw NPEs and some that prevented artillery weapons from being fired by airborne units.

Note that per errata, Artillery weapons should not be grouped in bays. This fix works properly when separate artillery bays are created for each artillery weapon and its ammunition. It will waste ammo and only resolve a single shot impact if used if used with a bay of multiple weapons. Pending work from Hammer and Neoancient will change the unit files to resolve this.

### CODE REVIEW

File: megamek/src/megamek/common/weapons/ArtilleryBayWeaponIndirectFireHandler.java [hunk 1]
```diff
@@ -86,6 +86,33 @@ public boolean cares(IGame.Phase phase) {
...
+    @Override
+    protected void useAmmo() {
+        for (int wId : weapon.getBayWeapons()) {
+            Mounted bayW = ae.getEquipment(wId);
+            // check the currently loaded ammo
+            Mounted bayWAmmo = bayW.getLinked();
+
+            if (bayWAmmo == null) {// Can't happen. w/o legal ammo, the weapon
+                // *shouldn't* fire.
+                System.out.println("Handler can't find any ammo!  Oh no!");
```
Comment (NickAragua) [inline megamek/src/megamek/common/weapons/ArtilleryBayWeaponIndirectFireHandler.java]: You will want to log this instead of using System.out.println

Comment (mkerensky) [reply megamek/src/megamek/common/weapons/ArtilleryBayWeaponIndirectFireHandler.java]: This is a spot of code I found and repurposed where logging hasn't been changed yet. What's the new format supposed to be?  Is it the Try {   Catch ?  I've seen a few of those and I think I could change to use that.

Comment (NickAragua) [reply megamek/src/megamek/common/weapons/ArtilleryBayWeaponIndirectFireHandler.java]: There should be a member variable called "game", which probably has a logger instance. You don't need a try-catch block or anything like that. Wildj79 provided some good examples in the contributors channel just a few minutes ago: https://megamek-public.slack.com/files/U2FKHG8EL/F7UG5SZD0/log4j_example.java

**Comment by NickAragua** [comment]
WeaponAttackAction really needs to have its text tokenized for translation purposes, but that can be handled at another time.

**Pull request merged**